src/hooks/useENS.ts

```diff
@@ -0,0 +1,25 @@
+import { ethers } from 'ethers';
+import { useEffect, useState } from 'react';
+import { EthNetworks } from 'services/web3/types';
+import { buildAlchemyUrl } from 'services/web3/wallet/connectors';
+
+const useENS = (address: string) => {
+  const provider = new ethers.providers.JsonRpcProvider(
+    buildAlchemyUrl(EthNetworks.Mainnet)
+  );
+  const [ensName, setENSName] = useState<string | null>(null);
```

Comment:
```suggestion
import { getProvider } from 'services/web3';
import { EthNetworks } from 'services/web3/types';

const useENS = (address: string) => {
  const provider = getProvider(EthNetworks.Mainnet);
  const [ensName, setENSName] = useState<string | null>(null);
```

Reply:
@pingustar take a look at this 🤗  to use `getProvider`, this works locally

Reply:
@bigint thanks! Looks better!

I am not very familiar with ENS, so please excuse this question, but how is this handled on ropsten? I think we need to handle network changes and set the provider accordingly.

Reply:
@pingustar no worries man, ENS is supported only in Mainnet unfortunately, they use Rinkeby but it is not an audited one, IMO all platform uses mainnet for ens resolving 🙇🏻

Reply:
@pingustar I see ens names are not resolving in https://79fe7172.phoenix-5t3.pages.dev/

If you like apply the suggested changes and we can test with that 😄 🤞🏻

Reply:
changes are committed, preview link will be up in a few minutes @bigint